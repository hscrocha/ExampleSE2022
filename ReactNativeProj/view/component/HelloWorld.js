import React from "react";
import { Text, View } from "react-native";
import MainStyle from "../MainStyle.style";


export default function HelloWorld(props){
    return (
        <View style={MainStyle.centerView}>
            <Text style=
            {[MainStyle.emphasisText,{color:props.mycolor}]}>
                Hello {props.children}
            </Text>
        </View>
    );
}