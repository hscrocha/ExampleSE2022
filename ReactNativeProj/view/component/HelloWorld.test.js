import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

test('HelloWorld Snapshot Test',function(){
    const element = renderer.create(
       <HelloWorld mycolor="green">Hello!</HelloWorld>
    );
    expect(element).toMatchSnapshot();
});
