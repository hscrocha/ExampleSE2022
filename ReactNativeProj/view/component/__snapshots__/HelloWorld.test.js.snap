// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 36,
          "textAlign": "center",
        },
        Object {
          "color": "green",
        },
      ]
    }
  >
    Hello 
    Hello!
  </Text>
</View>
`;
